src/ansys/fluent/core/quantity.py

```diff
@@ -0,0 +1,732 @@
+
+    def __rmul__(self, other):
+        return self.__mul__(other)
+
+    def __truediv__(self, other):
+        if isinstance(other, Quantity):
+            temp_value = self._si_value / other._si_value
+            temp_unit = self._get_si_unit(other, lambda x, y: x - y)
+            return Quantity(temp_value, temp_unit)
+        elif isinstance(other, int) or isinstance(other, float):
+            temp = Quantity(self._si_value / other, self._si_unit)
+            return temp
+
+    def __rtruediv__(self, other):
+        if not isinstance(other, Quantity) and isinstance(other, float):
+            return Quantity(other / self._si_value, self._si_unit)
+        else:
+            return other / self
+
+    def __add__(self, other):
+        if isinstance(other, Quantity):
+            if self.get_dimensions_list() == other.get_dimensions_list():
+                temp_value = float(self) + float(other)
+            else:
+                raise ValueError("Incompatible dimensions.")
+        elif isinstance(other, (float, int)):
+            if self.is_dimension_less():
+                temp_value = self._si_value + other
+            else:
+                raise ValueError("Incompatible dimensions.")
+        else:
+            return super().__add__(other)
+        return Quantity(temp_value, self._si_unit)
+
+    def __radd__(self, other):
+        return super().__add__(other)
+
+    def __sub__(self, other):
+        if isinstance(other, Quantity) and (
+            self.get_dimensions_list() == other.get_dimensions_list()
+        ):
+            temp_value = self._si_value - other._si_value
+        elif (
+            not isinstance(other, Quantity)
+            and self.is_dimension_less()
+            and (isinstance(other, int) or isinstance(other, float))
+        ):
+            temp_value = self._si_value - other
+        else:
+            raise ValueError("Incompatible dimensions.")
+        return Quantity(temp_value, self._si_unit)
+
+    def __rsub__(self, other):
+        return Quantity(other, "") - self
+
+    def __eq__(self, other):
+        if isinstance(other, Quantity):
+            return self._si_value == other._si_value and self._si_unit == other._si_unit
+        elif (
+            self.is_dimension_less()
+            and not isinstance(other, Quantity)
+            and isinstance(other, float)
+        ):
+            return float(self) == other
+
+    def __neg__(self):
+        return Quantity(-self.value, self.unit)
+
+    def __ge__(self, other):
+        if (
+            isinstance(other, Quantity)
+            and self.get_dimensions_list() == other.get_dimensions_list()
+            and self._si_unit == other._si_unit
+        ):
+            return float(self) > float(other)
+        elif (
+            self.is_dimension_less()
+            and (not isinstance(other, Quantity))
+            and isinstance(other, (float, int))
+        ):
```

Comment:
If dimensions don't match, you should throw an error.

Reply:
Maybe in the next PR, you can include all comparison functions (__gt__, __ge__, __lt__, __le__, __eq__, __neq__) and their __r* counterparts (__rgt__, etc.)

Reply:
> If dimensions don't match, you should throw an error.

Yes, Thank you.

Reply:
> Maybe in the next PR, you can include all comparison functions (**gt**, **ge**, **lt**, **le**, **eq**, **neq**) and their __r* counterparts (**rgt**, etc.)

Yes sure, we will add remaining functions in next PR.